module/input/ImageDecompressor/src/ImageDecompressor.h

```diff
@@ -0,0 +1,57 @@
+#ifndef MODULE_INPUT_IMAGEDECOMPRESSOR_H
+#define MODULE_INPUT_IMAGEDECOMPRESSOR_H
+
+#include <atomic>
+#include <cstdint>
+#include <memory>
+#include <mutex>
+#include <nuclear>
+
+#include "decompressor/DecompressorFactory.h"
+
+namespace module {
+namespace input {
+
+    class ImageDecompressor : public NUClear::Reactor {
+    private:
+        struct DecompressorContext {
+            struct Decompressor {
+                /// This atomic bool gets turned on/off depending on if it is in use
+                std::unique_ptr<std::atomic<bool>> active;
```

Comment:
Does it really need to be a unique pointer?

Reply:
It does have to be a unique pointer due to the way atomics work